File: englising-be/src/main/java/org/englising/com/englisingbe/word/entity/Word.java
package org.englising.com.englisingbe.word.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "word") // 이거 하던 중
public class Word {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "word_id")
    private Long wordId;

    @Column(name = "en_text")
    private String enText;

    @Column(name = "ko_text_1")
    private String koText1;

    @Column(name = "ko_text_2")
    private String koText2;

    @Column(name = "ko_text_3")
    private String koText3;

    @Column(name = "example")
    private Integer example;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;
}

File: englising-data/model/word.py
from sqlalchemy import Column, ForeignKey, Integer, String, Float, Boolean, TIMESTAMP, DECIMAL, Text, Date, Table
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from model.base import Base


class Word(Base):
    __tablename__ = 'word'
    word_id = Column(Integer, primary_key=True)
    en_text = Column(String(255), nullable=False)
    ko_text = Column(String(255), nullable=True)
    example = Column(String(512), nullable=False)
    created_at = Column(TIMESTAMP, default=func.now(), nullable=False)
    updated_at = Column(TIMESTAMP, onupdate=func.now())
    ko_text_1 = Column(String(255), nullable=True)
    ko_text_2 = Column(String(255), nullable=True)
    ko_text_3 = Column(String(255), nullable=True)

    track_words = relationship("TrackWord", back_populates="word")
